examples/qt/lineplot.py

```diff
@@ -0,0 +1,317 @@
+"""
+Lineplot Qt
+===========
+
+Complex example for lineplot in PyQt that displays 3 traces.
+The plot is a standard black on white with a legend on the right.
+
+"""
+import sys
+import numpy as np
+import time
+from math import pi, cos, sin, ceil, log10
+
+import fastplotlib as fpl
+
+try:
+    from PyQt6.QtWidgets import QApplication, QWidget, QVBoxLayout, QMainWindow
+    from PyQt6.QtCore import QTimer, Qt
+
+except ImportError:
+    from PyQt5.QtWidgets import QApplication, QWidget, QVBoxLayout, QMainWindow
+    from PyQt5.QtCore import QTimer, Qt
+
+def rotate(angle, axis_x, axis_y, axis_z):
+    """
+    Quaternion representing rotation around the given axis by the given angle.
+    """
+    a2 = angle/2.0
+    c = cos(a2)
+    s = sin(a2)
+    return (axis_x * s, axis_y * s, axis_z * s, c)
```

Comment:
nice, would be useful to have this in the addition to the existing [`Graphic.rotate`](https://github.com/fastplotlib/fastplotlib/blob/220527e0852c3ec337418805d5853c8967624865/fastplotlib/graphics/_base.py#L424-L444)